client/app/components/dynamic-form/DynamicForm.jsx

```diff
@@ -51,9 +56,14 @@ class DynamicForm extends React.Component {
   constructor(props) {
     super(props);
 
+    const hasFilledExtraField = some(props.fields, (field) => {
+      const { extra, initialValue } = field;
+      return extra && (!isEmpty(initialValue) || isNumber(initialValue) || isBoolean(initialValue) && initialValue);
```

Comment:
@ranbena this seemed the simplest without separating this and deep evaluating the values with their defaults. I've also tried the last one and my opinion is that it's an unnecessary extra complexity.